src/backend/distributed/connection/connection_configuration.c

```diff
@@ -234,6 +235,9 @@ GetConnParams(ConnectionHashKey *key, char ***keywords, char ***values,
 	 */
 	char nodePortString[12] = "";
 
+	StringInfo applicationName = makeStringInfo();
+	appendStringInfo(applicationName, "citus-%ld", GetGlobalPID());
```

Comment:
lets use `CITUS_APPLICATION_NAME_PREFIX`

Reply:
have we considered other formats? (e.g. `"citus gpid=%ld"`)

Reply:
Note that `application_name` has 63 chars limit:
```C
		len = pg_mbcliplen(ident, len, NAMEDATALEN - 1);
```

If you set it  > 63, PG will truncate automatically
```SQL
set application_name TO '1234567891234567891234567891234567891234567891234567891234567891';
NOTICE:  identifier "1234567891234567891234567891234567891234567891234567891234567891" will be truncated to "123456789123456789123456789123456789123456789123456789123456789"
SET
```

max `uint64` is `18446744073709551615`, which is just 20 chars. So, Marco's suggestion sounds a lot nicer and seems to work fine.